libs/ui/src/hooks/use_usb_drive.ts

```diff
@@ -1,14 +1,48 @@
 import useInterval from 'use-interval';
-import { sleep, usbstick } from '@votingworks/utils';
+import { assert, sleep, usbstick } from '@votingworks/utils';
 import { useCallback, useRef, useState } from 'react';
 import { LogEventId, Logger, LoggingUserRole } from '@votingworks/logging';
 import { useCancelablePromise } from './use_cancelable_promise';
 
 export const POLLING_INTERVAL_FOR_USB = 100;
 export const MIN_TIME_TO_UNMOUNT_USB = 1000;
 
+function isFat32(usbDriveInfo: KioskBrowser.UsbDriveInfo) {
+  return usbDriveInfo.fsType === 'vfat' && usbDriveInfo.fsVersion === 'FAT32';
+}
+
+const CASE_INSENSITIVE_ALPHANUMERICS = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+
+function randomCaseInsensitiveAlphanumeric(): string {
+  return CASE_INSENSITIVE_ALPHANUMERICS[
+    Math.floor(Math.random() * CASE_INSENSITIVE_ALPHANUMERICS.length)
+  ];
+}
+
+const VX_USB_LABEL_REGEXP = /VxUSB-[A-Z0-9]{5,}/i;
+
+function generateVxUsbLabel(usbDriveInfo: KioskBrowser.UsbDriveInfo): string {
+  const oldLabel = usbDriveInfo.label;
+  if (oldLabel && VX_USB_LABEL_REGEXP.test(oldLabel)) {
+    return oldLabel;
+  }
+
+  let newLabel = 'VxUSB-';
+
+  for (let i = 0; i < 5; i += 1) {
+    newLabel += randomCaseInsensitiveAlphanumeric();
+  }
+
+  return newLabel;
+}
+
+export interface PostFormatUsbOptions {
+  action: 'mount' | 'eject';
+}
+
 export type UsbDriveStatus =
   | 'absent'
+  | 'bad_format'
   | 'mounting'
   | 'mounted'
   | 'ejecting'
```

Comment:
We could also have `formatting` state but I did not want to handle that in all the frontends.

Reply:
Yeah, that seems fine. It shouldn't be possible to get into that state other than in the code that runs the format operation.